hibernate-core/src/main/java/org/hibernate/boot/model/internal/AnnotationBinder.java

```diff
@@ -800,6 +806,15 @@ private static JdbcMapping resolveJavaType(Class<JavaType<?>> type, MetadataBuil
 		return typeConfiguration.getBasicTypeRegistry().resolve( jtd, jdbcType );
 	}
 
+	private static ManagedBean<? extends Supplier> resolveParamResolver(ParamDef paramDef, MetadataBuildingContext context) {
+		Class<? extends Supplier> clazz = paramDef.resolver();
+		if (clazz.isInterface()) {
```

Comment:
It's totally valid to use an interface as bean type. The only case when we should skip trying to create a bean is when the annotation default is used.

```suggestion
		if ( clazz == Supplier.class ) {
```